### ISSUE TITLE
Redis Cluster SEGFAULT on 2nd get request

### ISSUE THREAD

**Issue by linxlad** [phpredis/phpredis#1129]
So I have tried both 3.1 and 3.1 RC 2 .DLL's.

The first get request is fine and then a perform a write and that's fine too. When it comes to any subsequent get/exists anything to retrieve my PHP script stops executing and windows event log records this:
```
Faulting application name: php-cgi.exe, version: 7.1.2.0, time stamp: 0x58a379a5
Faulting module name: php7ts.dll, version: 7.1.2.0, time stamp: 0x58a37ece
Exception code: 0xc0000005
Fault offset: 0x0001e438
Faulting process id: 0x10b0
Faulting application start time: 0x01d2998a7aa72f61
Faulting application path: C:\Program Files (x86)\DevDesktop\php7_1\php-cgi.exe
Faulting module path: C:\Program Files (x86)\DevDesktop\php7_1\php7ts.dll
Report Id: 88c96d22-e8cf-4dd2-a757-d08d371402c4
Faulting package full name: 
Faulting package-relative application ID: 
```

```
$cluster = new \RedisCluster(null, [
 // Hosts listed here.
]);

$cluster->setOption(\RedisCluster::OPT_SLAVE_FAILOVER, \RedisCluster::FAILOVER_ERROR);
```

Anyone experience this, I run xdebug at the time?

**Comment by michael-grunder** [comment]
Hi,

This is clearly an issue because we've had a couple of other similar reports.  It always seems to be on the second call.  The other report was for hgetall but it seems that isn't relevant.

Are you able to replicate the behavior with a small script or does it only happen very rarely?  Another thing that would be helpful is stack trace.

Interestingly, the other report involved using xdebug as well, so that might be a clue.  You might want to try without xdebug to see if the problem goes away.

Once I can replicate the problem it should be easy to fix, but I haven't been able to l.

**Comment by linxlad** [comment]
Thanks @michael-grunder I am not sure where I would get the stack trace from or how to produce one?

It happens on every 2nd get method call. Will try it without xdebug. I basically created a SessionHandler in drupal 8 to override their own session handler but we use clustered redis here shared with the Java apps.

**Comment by linxlad** [comment]
Just tried it with xdebug disabled and same issue.

**Comment by linxlad** [comment]
Tried it on a Debian box with the same outcome.

**Comment by michael-grunder** [comment]
Can you post a small script that causes the issue as well as the exact
version of php and redis on the Debian box?  You can omit server ips
obviously.

In the morning I can test against that exact script on the same version of
php and redis on my ubuntu machine.

Since you have a Debian box you can try running:
USE_ZEND_ALLOC=0 php <your script>

If it's a double free this should show that.

**Comment by michael-grunder** [comment]
Actually I'm sorry.  I reread your initial issue.  You're running the version from pecl?  This may have been fixed here:
https://github.com/phpredis/phpredis/commit/162d8820bc6e179419fe5db593ec47122e8d6129

On Debian can you try running the current develop branch?  If that solves it I can get that on pecl later today.

**Comment by linxlad** [comment]
Ah sweet, is there a .DLL with this fix in yet?

**Comment by linxlad** [comment]
Also just checked the Debian box and the .so built is based on the cluster file with those fixes :(

**Comment by michael-grunder** [comment]
The Debian box is running from develop?  This change isn't on pecl.

In order to diagnose this I'll need a small script that causes the crash
for you.  If it always crashes on the second cal it'll likely only be a few
lines (connect, then two caps)

That way I can know I'm executing code that crashes for you.

Did you try running with USE_ZEND_ALLOC=0 ?

**Comment by michael-grunder** [comment]
If you haven't yet solved the problem yet you can always try 3.1.2 (which is presently up on here and on PECL).  I have not been able to recreate this issue with any newer version.
